### PR TITLE
Checking for both types of region/zone labels

### PR DESCRIPTION
**Pull request by bathina2**
## Change Overview

In kubernetes 1.17 they are deprecating the "failure-domain..." labels and moving towards "topology..."
This change is checks for both labels when deriving the zone and the region.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [x] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues

- #XXX

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/kube/utils.go [hunk 1]
```diff
@@ -83,3 +95,44 @@ func GetPodContainerFromStatefulSet(ctx context.Context, cli kubernetes.Interfac
...
+}
+
+func GetZoneFromPV(pv v1.PersistentVolume) string {
+	return GetZoneFromLabels(pv.Labels)
+}
+
+func GetZoneFromLabels(labels map[string]string) string {
+	var zone string
+	if v, ok := labels[FDZoneLabelName]; ok {
+		zone = v
```
Comment (tdmanv) [inline pkg/kube/utils.go]: [nit] just return v in these statements

File: pkg/kube/utils.go [hunk 2]
```diff
@@ -83,3 +95,44 @@ func GetPodContainerFromStatefulSet(ctx context.Context, cli kubernetes.Interfac
...
+
+func GetZoneFromNode(node v1.Node) string {
+	return GetZoneFromLabels(node.Labels)
+}
+
+func GetZoneFromPV(pv v1.PersistentVolume) string {
+	return GetZoneFromLabels(pv.Labels)
+}
+
+func GetZoneFromLabels(labels map[string]string) string {
```
Comment (tdmanv) [inline pkg/kube/utils.go]: Just confirming - these are labels, not annotations, right?

File: pkg/kube/utils.go [hunk 3]
```diff
@@ -83,3 +95,44 @@ func GetPodContainerFromStatefulSet(ctx context.Context, cli kubernetes.Interfac
...
+	if v, ok := labels[FDZoneLabelName]; ok {
+		zone = v
+	} else if v, ok := labels[TopologyZoneLabelName]; ok {
+		zone = v
+	}
+	return zone
+}
+
+func GetRegionFromNode(node v1.Node) string {
+	return GetRegionFromLabels(node.Labels)
```
Comment (tdmanv) [inline pkg/kube/utils.go]: Just confirming - these are labels, not annotations, right?

File: pkg/kube/utils.go [hunk 4]
```diff
@@ -83,3 +95,44 @@ func GetPodContainerFromStatefulSet(ctx context.Context, cli kubernetes.Interfac
 	}
 	return podName, container[0].Name, nil
 }
+
+// Following functions get the regions and zones from
+// labels on Nodes and Persistent Volumes
+// As of kubernetes 1.17 the "failure.domain" annotation
+// has been deprecated in favor of the "topology" annotation
+
+func GetZoneFromNode(node v1.Node) string {
```
Comment (tdmanv) [inline pkg/kube/utils.go]: [note] this ship has sailed, but I prefer not prefixing with `Get`. see https://golang.org/doc/effective_go.html#Getters

File: pkg/kube/utils.go (hunk 2 above)
Comment (bathina2) [reply pkg/kube/utils.go]: yes these are labels

File: pkg/kube/utils.go (hunk 3 above)
Comment (bathina2) [reply pkg/kube/utils.go]: labels

File: pkg/kube/utils.go (hunk 4 above)
Comment (bathina2) [reply pkg/kube/utils.go]: Will avoid this in the future

**Pull request merged**